src/shared_definitions.hpp

```diff
@@ -259,17 +261,76 @@ using IndexSpec = seqan3::default_sdsl_index_type;
+template <typename TOrigSeqs, typename TOrigAlph, typename TTransAlph, typename TRedAlph>
+struct TTransSbjSeqsImpl
+{
+    using type = decltype(std::declval<TOrigSeqs&>() | seqan3::views::translate_join);
+};
+
+template <typename TOrigSeqs, typename TSameAlph>
+struct TTransSbjSeqsImpl<TOrigSeqs, TSameAlph, TSameAlph, seqan3::semialphabet_any<6>>
+{
+    using type = decltype(std::declval<TOrigSeqs&>() | seqan3::views::duplicate);
+};
+
+template <typename TOrigSeqs, typename TSameAlph, typename TRedAlph>
+struct TTransSbjSeqsImpl<TOrigSeqs, TSameAlph, TSameAlph, TRedAlph>
+{
+    using type = decltype(seqan3::views::type_reduce(std::declval<TOrigSeqs &>()));
+};
+
+// Translated query sequences
+template <typename TOrigSeqs, typename TOrigAlph, typename TTransAlph, typename TRedAlph>
+struct TTransQrySeqsImpl
+{
+    using type = decltype(std::declval<TOrigSeqs&>() | seqan3::views::translate_join);
+};
 
-template <typename TSpec, typename TAlph>
-using TRedAlphModString =
-  decltype(std::declval<TSpec &>() | seqan3::views::deep{seqan3::views::convert<TAlph>});
+template <typename TOrigSeqs, typename TRedAlph>
+    requires seqan3::nucleotide_alphabet<TRedAlph>
+struct TTransQrySeqsImpl<TOrigSeqs, seqan3::dna5, seqan3::dna5, TRedAlph>
+{
+    using type = decltype(std::declval<TOrigSeqs&>() | seqan3::views::add_reverse_complement);
+};
 
-template <typename TSpec, typename TAlph>
-using TRedNuclAlphModString =
-  decltype(std::declval<TSpec &>() | seqan3::views::dna_n_to_random<TAlph>);
+template <typename TOrigSeqs>
+struct TTransQrySeqsImpl<TOrigSeqs, seqan3::dna5, seqan3::dna5, seqan3::semialphabet_any<6>>
+{
+    using type = decltype(std::declval<TOrigSeqs&>() | seqan3::views::add_reverse_complement | seqan3::views::duplicate);
+};
+
+template <typename TOrigSeqs, typename TRedAlph>
+struct TTransQrySeqsImpl<TOrigSeqs, seqan3::aa27, seqan3::aa27, TRedAlph>
+{
+    using type = decltype(seqan3::views::type_reduce(std::declval<TOrigSeqs &>()));
+};
+
+// Reduced sequences
+template <typename TTransSeqs, typename TTransAlph, typename TRedAlph>
+struct TRedSeqsImpl
+{
+    using type = decltype(std::declval<TTransSeqs&>() | seqan3::views::deep{seqan3::views::convert<TRedAlph>});
+};
+
+template <typename TTransSeqs, typename TSameAlph>
+struct TRedSeqsImpl<TTransSeqs, TSameAlph, TSameAlph> // no reduction
+{
+    using type = decltype(seqan3::views::type_reduce(std::declval<TTransSeqs &>()));
+};
+
+template <typename TTransSeqs>
+struct TRedSeqsImpl<TTransSeqs, seqan3::dna5, seqan3::dna4>
+{
+    using type = decltype(std::declval<TTransSeqs &>() | seqan3::views::dna_n_to_random<seqan3::dna4>);
+};
+
+template <typename TTransSeqs>
+struct TRedSeqsImpl<TTransSeqs, seqan3::dna5, seqan3::semialphabet_any<6>>
+{
+    using type = decltype(std::declval<TTransSeqs &>() | seqan3::views::dna_n_to_random<seqan3::dna4>
+                                                       | seqan3::views::translate_bisulfite);
+};
```

Comment:
This solution is much better than the lazy-💩 we had before!

But maybe add one line of comment before everyone of these specialisations that documents the case?